### ISSUE TITLE
Where is Testing for Race Conditions (OWASP-AT-010) from OTGv3?

### ISSUE THREAD

**Issue by martinbydefault** [OWASP/wstg#571]
I was trying to map **Testing for Race Conditions (OWASP-AT-010)** from OTGv3 ( https://wiki.owasp.org/index.php/Testing_for_Race_Conditions_(OWASP-AT-010) ) to the latest version but I can't find any material regarding race conditions.

**Comment by kingthorin** [comment]
https://github.com/OWASP/wstg/blob/master/document/4-Web_Application_Security_Testing/06-Session_Management_Testing/08-Testing_for_Session_Puzzling.md

Looks like it was removed in 4.0: https://wiki.owasp.org/index.php/OWASP_Testing_Guide_v4_Table_of_Contents

**Comment by martinbydefault** [comment]
Yes I saw that 4.0 doesn't have it.
I don't get why this was removed, given that this is a very specific issue and there are even new techniques to exploit this, for example using BURP Suite Turbo Intruder plugin (https://portswigger.net/research/turbo-intruder-embracing-the-billion-request-attack)

**Comment by kingthorin** [comment]
I can't say why it might have been removed 8-10 years ago. If someone feels like reviving it I'm good with that. The 3.0 entry is awfully sparse on useful information.

**Comment by ThunderSon** [comment]
I will try and review the content (from v3) and cross reference it with other locations in the guide to see where it was ported. Pretty doubtful it has been removed, more so renamed or rewritten.
Should have an answer by early next week (unless I take a surprise vacation)

**Comment by ThunderSon** [comment]
Weird.
@MatOwasp might you know more on this. I was not able to find anything on this matter.

**Comment by MatOwasp** [comment]
Yes, I remember we decided to drop it because not really executed during a test.

**Comment by kingthorin** [comment]
Reference testbed:

- http://racetheweb.io/bank
- https://github.com/insp3ctre/race-the-web

**Comment by kingthorin** [comment]
@martinbydefault any interest in tackling this?

**Comment by martinbydefault** [comment]
@kingthorin Sorry for the delay in the answer. At the moment I'm not able to tackle this but will try to do it in the near future. I also suggested it in the pool of collaboration ideas of my local OWASP chapter to see if anyone wants to take it sooner than me.

**Comment by lightangel1412** [comment]
Hi everyone, 
Should we add this test case (Test case for Race Condition) into section "4.10 Business Logic Testing"?

**Comment by kingthorin** [comment]
That seems reasonable to me.

**Comment by kingthorin** [comment]
https://hackerone.com/reports/1540969

**Comment by 0xBassia** [comment]
I'd like to take this on. I'll add it as a new test case under Business Logic Testing, covering TOCTOU/race condition patterns in web applications: concurrent request abuse on financial endpoints, parallel session manipulation, coupon/reward multi-redemption, and file operation races. Will follow the existing WSTG format and have a PR up shortly.
